File: W2D4-CardViewAndRecyclerView/app/src/main/java/com/example/cardviewandrecyclerview/MainActivity.java
package com.example.cardviewandrecyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements CompanyAdapter.CompanyItemClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RecyclerView companyRecyclerView = findViewById(R.id.recyclerViewCompany);

        ArrayList<CompanyModel>  companyModelArrayList = new ArrayList<CompanyModel>();

        companyModelArrayList.add(new CompanyModel("Apple","1976",R.drawable.ic_apple));
        companyModelArrayList.add(new CompanyModel("Google","1998",R.drawable.ic_google));
        companyModelArrayList.add(new CompanyModel("Amazon","1994",R.drawable.ic_amazon));
        companyModelArrayList.add(new CompanyModel("Microsoft","1975",R.drawable.ic_microsoft));
        companyModelArrayList.add(new CompanyModel("Apple","1976",R.drawable.ic_apple));
        companyModelArrayList.add(new CompanyModel("Google","1998",R.drawable.ic_google));
        companyModelArrayList.add(new CompanyModel("Amazon","1994",R.drawable.ic_amazon));
        companyModelArrayList.add(new CompanyModel("Microsoft","1975",R.drawable.ic_microsoft));
        companyModelArrayList.add(new CompanyModel("Apple","1976",R.drawable.ic_apple));
        companyModelArrayList.add(new CompanyModel("Google","1998",R.drawable.ic_google));
        companyModelArrayList.add(new CompanyModel("Amazon","1994",R.drawable.ic_amazon));
        companyModelArrayList.add(new CompanyModel("Microsoft","1975",R.drawable.ic_microsoft));
        companyModelArrayList.add(new CompanyModel("Apple","1976",R.drawable.ic_apple));
        companyModelArrayList.add(new CompanyModel("Google","1998",R.drawable.ic_google));
        companyModelArrayList.add(new CompanyModel("Amazon","1994",R.drawable.ic_amazon));
        companyModelArrayList.add(new CompanyModel("Microsoft","1975",R.drawable.ic_microsoft));

        CompanyAdapter companyAdapter = new CompanyAdapter(this,companyModelArrayList,this);

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false);
        companyRecyclerView.setLayoutManager(linearLayoutManager);
        companyRecyclerView.setAdapter(companyAdapter);
    }
    @Override
    public void onItemClicked(CompanyModel companyModel) {

        Intent intent = new Intent(MainActivity.this ,SecondActivity.class);
        intent.putExtra("name",companyModel.getCompanyName());
        intent.putExtra("date", companyModel.getCompanyYear());
        intent.putExtra("image", companyModel.getCompanyLogo());
        intent.putExtra("company", companyModel);
        startActivity(intent);
    }
}

File: W7D1-Fragments/app/src/main/java/com/example/w7d1_fragments/MainActivity.kt
package com.example.w7d1_fragments

import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // MainActivity()
        // new MainActivity()
        val listFragment = ListFragment()
        val detailFragment = DetailFragment()


        supportFragmentManager.beginTransaction()
            .replace(R.id.full_container, listFragment)
            .commit()


        val myButton = findViewById<Button>(R.id.my_button)
        myButton.setOnClickListener {
            // fragment değiştirme kısmı ->

            // gönderilecek veri hazılanıyor
            val myBundle = Bundle()
            myBundle.putInt("my_counter", 10)

            detailFragment.arguments = myBundle

            // sayfa açılıyor
            supportFragmentManager.beginTransaction()
                .replace(R.id.full_container, detailFragment)
                .commit()

        }
    }
}